openwisp_monitoring/device/tests/test_api.py

```diff
@@ -433,6 +437,31 @@ def test_get_device_metrics_400_bad_timezone(self):
             self.assertEqual(r.status_code, 400)
             self.assertIn('Unkown Time Zone', r.data)
 
+    def test_device_metrics_received_signal(self):
+        d = self._create_device(organization=self._create_org())
+        dd = DeviceData(name='test-device', pk=d.pk)
+        self._create_object_metric(
+            name='ping', key='ping', field_name='reachable', content_object=d
+        )
+        with catch_signal(device_metrics_received) as handler:
+            response = self.client.get(self._url(d.pk.hex, d.key))
+        request = response.renderer_context['request']
+        handler.assert_called_once_with(
+            instance=dd,
+            request=request,
+            sender=DeviceData,
+            signal=device_metrics_received,
+        )
+
+    # Tests device status is set to ok if no related checks present
+    def test_status_set_ok(self):
```

Comment:
why is this test here? `test_api` should only test the API.
I'm also not 100% sure what's the aim of this test.

Reply:
It was stated in the issue
> If there's no related check, move the status to OK.

The above test was included to verify the same but yes, I have messed up the location of almost all functions which might make it difficult to relate the same

Reply:
ok, no problem, now I see more clearly, last night I was pretty tired from an entire day of review and I got confused regarding why this test was here. You may want to move all this functionality to a new test suite called `test_recovery.TestRecovery`.

Reply:
Done